src/java/htsjdk/samtools/reference/AbstractFastaSequenceFile.java

```diff
@@ -40,6 +42,7 @@
  */
 abstract class AbstractFastaSequenceFile implements ReferenceSequenceFile {
     protected final File file;
+    protected final Path pathFile;
```

Comment:
I'm a bit worried that this will cause confusing bugs in the future.  It's really unfortunate that `file` is a protected field used by subclasses instead of a property.  I think in this case it might be worth breaking the existing api to avoid having 2 parallel fields holding the same data.